packages/components/src/ui/menu/component.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { contextConnect, useContextSystem } from '@wp-g2/context';
+import { cx } from '@wp-g2/styles';
+// eslint-disable-next-line no-restricted-imports
+import { Menu as ReakitMenu } from 'reakit';
+
+/**
+ * Internal dependencies
+ */
+import { usePopoverResizeUpdater } from '../popover/utils';
+import { View } from '../view';
+import { MenuContext } from './context';
+import * as styles from './styles';
+
+/**
+ * `Menu` is an actionable component that displays a list of actions, links, or informative content.
+ *
+ * @example
+ * ```jsx
+ * <Menu>
+ *  <MenuItem>...</MenuItem>
+ *  <MenuItem>...</MenuItem>
+ *  <MenuItem>...</MenuItem>
+ * </Menu>
+ * ```
+ *
+ * @see https://reakit.io/docs/menu/#menu
+ *
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').Props, 'div'>} props
+ * @param {import('react').Ref<any>} forwardedRef
+ */
+function Menu( props, forwardedRef ) {
+	const { children, className, menu, ...otherProps } = useContextSystem(
+		props,
+		'Menu'
+	);
+
+	const resizeListener = usePopoverResizeUpdater( {
+		onResize: menu?.unstable_update,
```

Comment:
If we're going to use unstable features from packages, we should make sure we [use the exact version for that package on package.json](https://reakit.io/docs/experimental/#use-exact-version-in-packagejson) so as to avoid issues like #23332.